source/slang/slang-options.cpp

```diff
@@ -972,9 +976,10 @@ struct OptionsParser
                             String name;
                             SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, name));
 
-                            String slice = argStr.subString(1, index - 1);
+                            // Skip the initial -, up to the last -
+                            UnownedStringSlice passThruSlice(argStr.begin() + 1, argStr.begin() + index);
```

Comment:
It might be nice to add a `.subString()` method to `UnownedStringSlice`  so that the line could just change to:

```
auto slice = argStr.subString(1, index - 1);
```

and thus be independent of whether `argStr` is a `String` or `UnownedStringSlice` or whatever else we might define down the line.